store/tikv/rawkv.go

```diff
@@ -306,11 +306,63 @@ func (c *RawKVClient) Scan(startKey []byte, limit int) (keys [][]byte, values []
 	return
 }
 
-func (c *RawKVClient) sendReq(key []byte, req *tikvrpc.Request) (*tikvrpc.Response, *KeyLocation, error) {
+// ReverseScan queries continuous kv pairs in range [endKey, startKey), up to limit pairs.
+// Direction is different from Scan, upper to lower.
+// If endKey is empty, it means unbounded.
+// If you want to include the startKey or exclude the endKey, append a '\0' to the key. For example, to scan
+// (endKey, startKey], you can write:
+// `ReverseScan(append(startKey, '\0'), append(endKey, '\0'), limit)`.
+// It doesn't support Scanning from "", because locating the last Region is not yet implemented.
+func (c *RawKVClient) ReverseScan(startKey, endKey []byte, limit int) (keys [][]byte, values [][]byte, err error) {
+	start := time.Now()
+	defer func() {
+		metrics.TiKVRawkvCmdHistogram.WithLabelValues("raw_reverse_scan").Observe(time.Since(start).Seconds())
+	}()
+
+	if limit > MaxRawKVScanLimit {
+		return nil, nil, errors.Trace(ErrMaxScanLimitExceeded)
+	}
+
+	for len(keys) < limit {
+		req := &tikvrpc.Request{
+			Type: tikvrpc.CmdRawScan,
+			RawScan: &kvrpcpb.RawScanRequest{
+				StartKey: startKey,
```

Comment:
forget to set `Endkey`?

Reply:
The original PR is merged after another PR that supports EndKey, which is not cherry-picked to release-2.1. Here the EndKey was removed manually.

Reply:
Tests about EndKey is also removed.

Reply:
should we cherry pick that pr to release 2.1 as well?

Reply:
I don't think so. There are no requirement of that feature in release 2.1, but cherry-picking it needs cherry-picking another PR in kvproto and another PR in TiKV. Raw reverse scan is not required too. The reason of cherry-picking this PR is that reverse querying support of RegionCache is needed.

Reply:
okay